supportFiles/maps.js

```diff
@@ -0,0 +1,235 @@
+let map = [];
+const createShip = () => {
+    for (let x = 0; x < 4; x++) {
+    				map[x] = [];
+        for (let y = 0; y < 3; y++) {
+            addRoomShip(x, y); 
+        }
+    }
+}
+const addRoomShip = (x, y) => {
+    map[x][y] = roomShip(x,y);
+}
+const roomShip = (x,y) => {
```

Comment:
Think about having rooms in an array and populating a matrix through iteration. Or more fun, shuffle the array and then do it. Then each game could have a different ship layout.